### PR TITLE
[tune] tf2.0 testing and supporting callables

### PR DESCRIPTION
**Pull request by richardliaw**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.

### CODE REVIEW

File: python/ray/tune/experiment.py [hunk 1]
```diff
@@ -160,16 +160,11 @@ def _register_if_needed(cls, run_object):
...
-                    "Not auto-registering lambdas - resolving as variant.")
-                return run_object
-            else:
-                name = run_object.__name__
-                register_trainable(name, run_object)
-                return name
-        elif isinstance(run_object, type):
+        elif run_object.__name__ == "<lambda>":
+            logger.warning(
+                "Not auto-registering lambdas - resolving as variant.")
```
Comment (ericl) [inline python/ray/tune/experiment.py]: Do we still allow this? I think you need an explicit tune.sample_from() right?

**Comment by richardliaw** [comment]
Yeah good point, I’ll fix.

File: python/ray/tune/experiment.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/experiment.py]: oh wait; this is for the `tune.run(lambda x, reporter: reporter(4))` and not for the space.

Comment (richardliaw) [reply python/ray/tune/experiment.py]: OK actually I get what you mean now; fixed.

**Review by ericl — APPROVED** [review]

**Pull request merged**